ush/get_obs.py

```diff
@@ -0,0 +1,1004 @@
+    ----------
+    For NDAS, the archive interval is 6 hours, i.e. the obs files are
+    bundled into 6-hourly archives.  The archives are organized such that
+    each one contains 7 files (not say 6).  The archive associated with
+    time yyyymmddhh_arcv contains the hourly files at
+
+      yyyymmddhh_arcv - 6 hours
+      yyyymmddhh_arcv - 5 hours
+      ...
+      yyyymmddhh_arcv - 2 hours
+      yyyymmddhh_arcv - 1 hours
+      yyyymmddhh_arcv - 0 hours
+
+    These are known as the tm06, tm05, ..., tm02, tm01, and tm00 files,
+    respectively.  Thus, the tm06 file from the current archive, say the
+    one associated with time yyyymmddhh_arcv, has the same valid time as
+    the tm00 file from the previous archive, i.e. the one associated with
+    time (yyyymmddhh_arcv - 6 hours).  It turns out that the tm06 file from
+    the current archive contains more/better observations than the tm00
+    file from the previous archive.  Thus, for a given archive time
+    yyyymmddhh_arcv, we use 6 of the 7 files at tm06, ..., tm01 but not
+    the one at tm00, effectively resulting in 6 files per archive for NDAS
+    obs.  The obs availability interval is then
+
+      obs_avail_intvl_hrs = (24 hrs)/[(4 archives)*(6 files/archive)]
+                          = 1 hr/file
+
+    i.e. there is one obs file for each hour of the day containing values
+    at that hour.  The archive corresponding to hour 0 of the current day
+    contains 6 files valid at hours 18 through 23 of the previous day.  The
+    archive corresponding to hour 6 of the current day contains 6 files
+    valid at hours 0 through 5 of the current day, and the archives
+    corresponding to hours 12 and 18 of the current day each contain 6
+    files valid at hours 6 through 11 and 12 through 17 of the current day.
+    Thus, to obtain all the hourly values for the current day (from hour
+    0 to hour 23), we must extract the 6 obs files (excluding the tm00
+    ones) from the three archives corresponding to hours 6, 12, and 18 of
+    the current day and the archive corresponding to hour 0 of the next
+    day.  This corresponds to an archive hour sequence set below of [6, 12,
+    18, 24].  Thus, in the simplest case in which the observation retrieval
+    times include all hours of the current task's day at which obs files
+    are available and none of the obs files for this day already exist on
+    disk, this sequence will be [6, 12, 18, 24].  In other cases, the
+    sequence we loop over will be a subset of [6, 12, 18, 24].
+    """
+
+    # Convert obtype to upper case to simplify code below.
+    obtype = obtype.upper()
+
+    # For convenience, get the verification portion of the configuration
+    # dictionary.
+    vx_config = cfg['verification']
+
+    # Get the time interval (in hours) at which the obs are available.
+    key = obtype + '_OBS_AVAIL_INTVL_HRS'
+    obs_avail_intvl_hrs = vx_config[key]
```

Comment:
There are a lot of definitions here for `key` used only once and then over-written; this can lead to very hard-to-track errors down the road, let's just eliminate duplicate code

```suggestion
    obs_avail_intvl_hrs = vx_config[obtype + '_OBS_AVAIL_INTVL_HRS']
```

Reply:
Changed.